### PR TITLE
Optimise social feed update

### PR DESCRIPTION
**Pull request by ianopolous**

### CODE REVIEW

File: src/peergos/shared/user/TofuKeyStore.java [hunk 1]
```diff
@@ -48,6 +48,13 @@ public TofuKeyStore() {
         return chains.getOrDefault(username, expired.getOrDefault(username, Collections.emptyList()));
     }
 
+    /**
```
Comment (kevodwyer) [inline src/peergos/shared/user/TofuKeyStore.java:51]: update the javadocs

File: src/peergos/shared/user/TofuKeyStore.java [hunk 2]
```diff
@@ -57,6 +64,8 @@ public TofuKeyStore() {
                     // we are allowing expired chains to be stored
 
                     List<UserPublicKeyLink> existing = getChain(username);
+                    if (! existing.isEmpty() && existing.get(existing.size() - 1).equals(tail.get(tail.size() - 1)))
```
Comment (kevodwyer) [inline src/peergos/shared/user/TofuKeyStore.java:67]: not sure i understand this

Comment (ianopolous) [reply src/peergos/shared/user/TofuKeyStore.java:67]: This is just saying if nothing has changed for that user in the pki then we don't need to commit the tofukeystore.

File: src/peergos/shared/user/TofuKeyStore.java (hunk 1 above)
Comment (ianopolous) [reply src/peergos/shared/user/TofuKeyStore.java:51]: What do you mean? The critical thing here is the return value meaning.

**Review by kevodwyer — APPROVED** [review]

**Pull request merged**